### PR TITLE
Generalise line2route and add functions for creating od data from points

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

**Comment by Robinlovelace** [comment]
@richardellison can you review and comment/merge please.

Your thoughts on the radiation model are especially appreciated.

Robin

File: R/radiate.R [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    for(j in 1:nrow(p)){
+      if(i == j) next()
+      m <- p[[pop_var]][i]
+      n <- p[[pop_var]][j]
+      # create circle the radius of which is the distance between i and j centered on i
+      s_circle <- stplanr::buff_geo(p[i,], width = geosphere::distHaversine(p[i,], p[j,]))
+      ps <- p[-c(i, j),][s_circle,]
+      s <- sum(ps[[pop_var]])
+      l$flow[l$O == p@data[i,1] & l$D == p@data[j,1]] <-
+        p$population[i] * proportion * m * n / ((m + s) * (m + n + s))
```
Comment (richardellison) [inline R/radiate.R]: I think this should be:
`p$population[i] * proportion * ((m * n) / ((m + s) * (m + n + s)))`

**Comment by richardellison** [comment]
@Robinlovelace , it seems like just a minor error causing the discrepancies (see my note in the code). Can you please fix it and I'll merge.

**Comment by Robinlovelace** [comment]
@richardellison job done I believe.

**Comment by richardellison** [comment]
Looks good to me and the results make more sense now. Merging...

**Pull request merged**